pkg/lib/localstore/objectstore/queries.go

```diff
@@ -23,6 +24,32 @@ func (s *dsObjectStore) Query(q database.Query) ([]database.Record, error) {
 	return recs, err
 }
 
+func (s *dsObjectStore) QueryWithContext(ctx context.Context, q database.Query) (records []database.Record, err error) {
+	if err = ctx.Err(); err != nil {
+		return
+	}
+
+	recordsCh := make(chan []database.Record, 1)
+	errCh := make(chan error, 1)
+	go func() {
+		recs, err := s.Query(q)
+		if err != nil {
+			errCh <- err
+		} else {
+			recordsCh <- recs
+		}
+	}()
+
+	select {
+	case <-ctx.Done():
```

Comment:
I would actually do it synchronously, I dont like hanging goroutine especially if you return when context is done.

Reply:
Done!